src/server/cards/base/library.js

```diff
@@ -1,11 +1,24 @@
 import Card from 'cards/Card';
-import { Set } from 'immutable';
+import DirtyModel, { trackDirty } from 'utils/DirtyModel';
+import Pile from 'utils/Pile';
 
 export default class Library extends Card {
-  static name = 'Library';
   static cost = 5;
   static types = new Set(['Action']);
+
+  @trackDirty(pile => (pile.size > 0 ? pile.last().id : null))
```

Comment:
1. `trackDirty` not necessary
2. keep variable scope as small as possible...  no point having this be a class property if it is only ever used inside `onPlay` and the state does not need to be preserved between calls